dom/data/data.js

```diff
@@ -2,35 +2,28 @@
 
 var domDataState = require("can-dom-data-state");
 var mutationDocument = require("../mutation-observer/document/document");
+var assign = require("can-assign");
+
+var domData = {
+	// count of distinct elements that have domData set
+	_elementSetCount: 0
+};
 
 var deleteNode = function() {
 	return domDataState.delete.call(this);
 };
 
-// count of distinct elements that have domData set
-var elementSetCount = 0;
-
 var cleanupDomData = function(node) {
 	// decrement count if node was deleted
-	elementSetCount -= deleteNode.call(node) ? 1 : 0;
+	domData._elementSetCount -= deleteNode.call(node) ? 1 : 0;
 
 	// remove handler once all domData has been cleaned up
-	if (elementSetCount === 0) {
+	if (domData._elementSetCount === 0) {
 		mutationDocument.offAfterRemovedNodes(cleanupDomData);
 	}
 };
 
-/**
- * @module {{}} can-util/dom/data/data data
- * @parent can-util/dom
- * @description Allows associating data as a key/value pair for a particular
- * DOM Node.
- *
- * ```js
- * var domData = require("can-util/dom/data/data");
- * ```
- */
-module.exports = {
+assign(domData, {
```

Comment:
Just a thought: since we only ever, even for testing, just want to read the `elementSetCount` value, why not make a getter function to read it? Like
```js
domData = {
  _getElementSetCount: function () { return elementSetCount; }, ...
}
```

The benefit being no one can mess up the counter if they wanted to, but also using a function allows us to in the future log a warning or something when this is used.

Just a thought though. I would like to see `assign` dropped since we don't really need it.